test/common/http/http2/metadata_encoder_decoder_test.cc

```diff
@@ -182,6 +182,33 @@ TEST_F(MetadataEncoderDecoderTest, TestMetadataSizeLimit) {
   cleanUp();
 }
 
+TEST_F(MetadataEncoderDecoderTest, TestMetadataSizeLimitIncremental) {
```

Comment:
Why did you decide to repeat `TestMetadataSizeLimit` test here too? If there is an issue around that check, we now have two tests that will fail in exactly the same way. I think this makes this test more fragile (it can fail for more than one reason) and harder to understand (tests more than one thing).

Reply:
Someone should really disentangle the encoder and decoder tests. In any case, I think I've resolved the issue you were criticizing.